pkg/kv/kvserver/store.go

```diff
@@ -1904,6 +1904,52 @@ func ReadStoreIdent(ctx context.Context, eng storage.Engine) (roachpb.StoreIdent
 	return ident, err
 }
 
+// loadAndReconcilesReplicas loads the Replicas present on this
+// store. It reconciles inconsistent state and runs validation checks.
+//
+// TODO(sep-raft-log): also load *uninitialized* Replicas.
+func loadAndReconcileReplicas(
+	ctx context.Context, eng storage.Engine,
+) (map[storage.FullReplicaID]*roachpb.RangeDescriptor, error) {
+	ident, err := ReadStoreIdent(ctx, eng)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Since the caller knows the store ID, could it be passed in instead of reading it here? (to preserve the old behaviour)

Is it to make this function more "stand-alone", or some other reason from follow-up PRs?

Reply:
Yeah, outside of performance critical code paths I have a tendency to make things stand-alone. For example, these methods will probably live in `kvstorage` soon, and will be unit tested there (with something like the datadriven test introduced in https://github.com/tbg/cockroach/commit/e6223a28fe935ab70479b0decb9704cdf9b53f1b#diff-57be82ecc91b85772e9d59c726c4cb5e044fd31931b94cc2c0e38f702b0006c3). That doesn't mean we can't pass the storeID in there too, but might as well have only one source of truth instead of two. We might find a reason to do it differently as the work progresses, I'm open to that if it happens.